### PR TITLE
[Feature] Netstandard support

### PR DESCRIPTION
**Pull request by robertcoltheart**
@jonnii taking your branch and shifting it forwards.

Now that netstandard 2.0 is here, should we just go for that version? It would mean that the serializers would be almost a drop-in in terms of porting, and we can drop some of the `#if NETSTANDARD1_6` pre-compiler stuff.

### CODE REVIEW

**Comment by robertcoltheart** [comment]
Appveyor is having all sorts of issues with this, FYI. I don't see an `appveyor.yml` though, so I'm guessing you've configured all the build steps on the site directly? Should probably convert it to an `appveyor.yml` so that I can fix 😄.

**Comment by jonnii** [comment]
> Now that netstandard 2.0 is here, should we just go for that version?

yeah, why not. 

> Appveyor is having all sorts of issues with this, FYI. I don't see an appveyor.yml though, so I'm guessing you've configured all the build steps on the site directly? Should probably convert it to an appveyor.yml so that I can fix :smile:.

I'll sort that out. 

@robertcoltheart i'll also give you commit bit.

**Comment by jonnii** [comment]
@robertcoltheart i'll take a look at this tonight.

File: src/Chinchilla.Integration/Chinchilla.Integration.csproj [hunk 1]
```diff
@@ -1,131 +1,34 @@
...
-      <HintPath>..\packages\Microsoft.Net.Http.2.2.18\lib\net45\System.Net.Http.Extensions.dll</HintPath>
-    </Reference>
-    <Reference Include="System.Net.Http.Primitives">
-      <HintPath>..\packages\Microsoft.Net.Http.2.2.18\lib\net45\System.Net.Http.Primitives.dll</HintPath>
-    </Reference>
-    <Reference Include="System.Net.Http.WebRequest" />
+    <PackageReference Include="Castle.Core" Version="4.1.1" />
+    <PackageReference Include="nunit" Version="3.8.1" />
+    <PackageReference Include="RabbitMq.Client" Version="5.0.1" />
+    <PackageReference Include="SimpleJson" Version="0.38.0" />
```
Comment (jonnii) [inline src/Chinchilla.Integration/Chinchilla.Integration.csproj]: i think we should just reference newtonsoft.json, tbh it's basically a standard dependency everywhere now and it'll save us some headaches.

File: src/Chinchilla.Specifications/Serializers/DefaultMessageTypeFactorySpecification.cs [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
-    {
-        [Subject(typeof(DefaultMessageTypeFactory))]
-        public class in_general : WithSubject<DefaultMessageTypeFactory>
-        {
-            It should_not_have_cached_factory_by_default = () =>
-                Subject.HasCachedFactory(typeof(IMagicMessage)).ShouldBeFalse();
-        }
+// namespace Chinchilla.Specifications.Serializers
+// {
+//     public class DefaultMessageTypeFactorySpecification
```
Comment (jonnii) [inline src/Chinchilla.Specifications/Serializers/DefaultMessageTypeFactorySpecification.cs]: still need to figure out what to do with this, if we're using newtonsoft i think we can elide it as that can be taken care of by the serializer.

**Comment by jonnii** [comment]
I've changed my mind on net standard 2.0, lets target the lowest version we can and also take a hard dependency on newtonsoft.json

**Comment by robertcoltheart** [comment]
Roger that :+1:

**Comment by robertcoltheart** [comment]
@jonnii I've targeted 1.6, all the reflection emit stuff is fine, just a couple of tweaks necessary. Happy to switch to newtonsoft.json at some point, but the simplejson stuff is ok at the moment? To switch would require a bit more effort. Also noticed we are using the `SimpleJson.cs` code within the libraries themselves, so no need to take a reference to the package.

Just need the appveyor build fixed up, or else I'll write an `appveyor.yml`.

File: src/Chinchilla.Specifications/Serializers/DefaultMessageTypeFactorySpecification.cs (hunk 2 above)
Comment (robertcoltheart) [reply src/Chinchilla.Specifications/Serializers/DefaultMessageTypeFactorySpecification.cs]: This is fixed using netstandard 1.6. See comment below.

File: build/package.logging.nlog.nuspec [hunk 3]
```diff
@@ -15,7 +15,7 @@
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <tags>rabbitmq management api http chinchilla</tags>
     <dependencies>
-      <dependency id="NLog" version="4.2.0" />
+      <dependency id="NLog" version="5.0.0-beta10" />
```
Comment (robertcoltheart) [inline build/package.logging.nlog.nuspec]: This makes me nervous at the moment. No fully fledged support for netstandard at this point, so had to rely on the pre-release. Worth waiting before merging this?

File: src/Chinchilla.Specifications/Serializers/DefaultMessageTypeFactorySpecification.cs (hunk 2 above)
Comment (robertcoltheart) [reply src/Chinchilla.Specifications/Serializers/DefaultMessageTypeFactorySpecification.cs]: Oops, need to uncomment this.

File: build/package.logging.nlog.nuspec (hunk 3 above)
Comment (jonnii) [reply build/package.logging.nlog.nuspec]: I think this is ok for now. Will be a netstandard one soon I think and we don't need to take this dep.

**Comment by jonnii** [comment]
@robertcoltheart i'd merge this and do appveyor config in a diff PR.

**Comment by jonnii** [comment]
@robertcoltheart you have commit bit now so you may click the button ^_^

**Pull request merged**